cloudigrade/account/tasks.py

```diff
@@ -44,60 +44,83 @@ def copy_ami_snapshot(arn, ami_id, source_region):
         image.save()
         raise AwsSnapshotEncryptedError
 
-    aws.add_snapshot_ownership(snapshot)
+    aws.add_snapshot_ownership(customer_snapshot)
 
-    new_snapshot_id = aws.copy_snapshot(snapshot_id, source_region)
-    create_volume.delay(ami_id, new_snapshot_id)
+    snapshot_copy_id = aws.copy_snapshot(customer_snapshot_id, source_region)
+    volume_information = {
+        'arn': arn,
+        'source_region': source_region,
+        'ami_id': ami_id,
+        'customer_snapshot_id': customer_snapshot_id,
+        'snapshot_copy_id': snapshot_copy_id
+    }
+    logger.info(_(
+        '{0}: customer_snapshot_id={1}, snapshot_copy_id={2}').format(
+        'copy_ami_snapshot',
+        volume_information['customer_snapshot_id'],
+        volume_information['snapshot_copy_id']))
+    create_volume.delay(volume_information)
 
 
 @retriable_shared_task
 @rewrap_aws_errors
-def create_volume(ami_id, snapshot_id):
+def create_volume(volume_information):
     """
     Create an AWS Volume in the primary AWS account.
 
     Args:
-        ami_id (str): The AWS AMI id for which this request originated
-        snapshot_id (str): The id of the snapshot to use for the volume
+        volume_information (dict): Meta-data about scanned instances (arn,
+            source_region, ami_id, customer_snapshot_id, snapshot_copy_id)
 
     Returns:
         None: Run as an asynchronous Celery task.
 
     """
     zone = settings.HOUNDIGRADE_AWS_AVAILABILITY_ZONE
-    volume_id = aws.create_volume(snapshot_id, zone)
+    volume_id = aws.create_volume(volume_information['snapshot_copy_id'], zone)
     region = aws.get_region_from_availability_zone(zone)
+    volume_information['volume_id'] = volume_id
+    volume_information['volume_region'] = region
 
-    enqueue_ready_volume.delay(ami_id, volume_id, region)
+    logger.info(_('{0}: volume_id={1}, volume_region={2}').format(
+        'create_volume',
+        volume_information['volume_id'],
+        volume_information['volume_region']))
+    enqueue_ready_volume.delay(volume_information)
 
 
 @retriable_shared_task
 @rewrap_aws_errors
-def enqueue_ready_volume(ami_id, volume_id, region):
+def enqueue_ready_volume(volume_information):
     """
     Enqueues information about an AMI and volume for later use.
 
     Args:
-        ami_id (str): The AWS AMI id for which this request originated
-        volume_id (str): The id of the volume to mount
-        region (str): The region the volume is being created in
+        volume_information (dict): Meta-data about scanned instances (arn,
+            source_region, ami_id, customer_snapshot_id, snapshot_copy_id,
+            volume_id, volume_region)
 
     Returns:
         None: Run as an asynchronous Celery task.
 
     """
-    volume = aws.get_volume(volume_id, region)
+    volume = aws.get_volume(
+        volume_information['volume_id'],
+        volume_information['volume_region'])
     aws.check_volume_state(volume)
```

Comment:
Next line would be a great time to kick off another new task with a `.delay()` that cleans up our copy of the snapshot, since at this point we know our volume is good.